### PR TITLE
mgr/snap-schedule: use the right way to check the result returned by…

### PR DESCRIPTION
**Pull request by Mer1997**
Fixes: https://tracker.ceph.com/issues/57138
Signed-off-by: Mer Xuanyi <xuanyi.meng@xtaotech.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
- Documentation (select at least one)
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/snap_schedule/fs/schedule.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def rm_schedule(cls,
                              (path,))
             row = cur.fetchone()
 
-            if len(row) == 0:
+            if not row:
```
Comment (tchaikov) [inline src/pybind/mgr/snap_schedule/fs/schedule.py:299]:
```suggestion
            if row is None:
```

better off being more explicit.

**Comment by tchaikov** [comment]
also, might want to change "mgr(snap-schedule)" to "mgr/snap-schedule".

File: src/pybind/mgr/snap_schedule/fs/schedule.py (hunk 1 above)
Comment (Mer1997) [reply src/pybind/mgr/snap_schedule/fs/schedule.py:299]: > better off being more explicit.

the same style in add_retention/rm_retention, do we need to change them all?

https://github.com/ceph/ceph/blob/4235e99e593f41c3a2fddb5b2040a23e0d5974ec/src/pybind/mgr/snap_schedule/fs/schedule.py#L348

https://github.com/ceph/ceph/blob/4235e99e593f41c3a2fddb5b2040a23e0d5974ec/src/pybind/mgr/snap_schedule/fs/schedule.py#L372

Comment (tchaikov) [reply src/pybind/mgr/snap_schedule/fs/schedule.py:299]: one thing at a time. you could have a dedicated cleanup commit in this PR.

**Comment by tchaikov** [comment]
@Mer1997 you need to use `git commit --amend` and `git push -f` to update the commit message.

File: src/pybind/mgr/snap_schedule/fs/schedule.py (hunk 1 above)
Comment (Mer1997) [reply src/pybind/mgr/snap_schedule/fs/schedule.py]: > one thing at a time. you could have a dedicated cleanup commit in this PR.

changed

**Comment by Mer1997** [comment]
jenkins retest this please

**Comment by Mer1997** [comment]
jenkins test make check arm64

**Comment by Mer1997** [comment]
jenkins test api

**Comment by Mer1997** [comment]
jenkins test windows

**Review by tchaikov — APPROVED** [review]

**Comment by Mer1997** [comment]
jenkins test api

**Comment by Mer1997** [comment]
jenkins test windows

**Comment by Mer1997** [comment]
@tchaikov Mind look at this pull request, how to merge it?

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
@Mer1997 we need to pass the test first. there are some flaky tests. but let's give it another try.

**Comment by Mer1997** [comment]
> @Mer1997 we need to pass the test first. there are some flaky tests. but let's give it another try.

Thanks, I've already tried this a few times, but they just failed, again and agin.

**Comment by Mer1997** [comment]
jenkins test make check arm64

**Comment by Mer1997** [comment]
jenkins test windows

**Comment by Mer1997** [comment]
jenkins test windows

**Comment by Mer1997** [comment]
jenkins test make check arm64

**Comment by Mer1997** [comment]
jenkins test windows

**Comment by Mer1997** [comment]
jenkins test windows

**Comment by tchaikov** [comment]
@Mer1997 hi Mer, currently the windows build is broken. so we can consider all tests passed now. and this PR is only pending on a rados qa run. sorry for the confusion.

**Comment by Mer1997** [comment]
> @Mer1997 hi Mer, currently the windows build is broken. so we can consider all tests passed now. and this PR is only pending on a rados qa run. sorry for the confusion.

Never mind

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2022-08-23_13:19:39-rados-wip-kefu-testing-2022-08-22-2243-distro-default-smithi/

failures tracked by

- https://tracker.ceph.com/issues/57256
- https://tracker.ceph.com/issues/57163
- https://tracker.ceph.com/issues/55789

**Pull request merged**